### PR TITLE
WF-330 Improve documentation

### PR DESCRIPTION
**Pull request by lx223**

### CODE REVIEW

File: apis/deployment_v1alpha1/Deployment.cs [hunk 1]
```diff
@@ -370,7 +373,7 @@ public sealed partial class Deployment : pb::IMessage<Deployment> {
     ///
     /// The worker flags that the deployment exposes at runtime.
     ///
-    /// Only applicable to running deployments.
+    /// Only applicable to running deployments. It is omitted for deployments of other states.
```
[the remark this answers is not in the document]
Comment (lx223) [reply apis/deployment_v1alpha1/Deployment.cs]: Hey, these docs are automatically generated from the proto comments. So for changes to stick, we'll have to make changes over there. 

I am alright with either. @anne-edwards?

[the remark this answers is not in the document]
Comment (anne-edwards) [reply apis/deployment_v1alpha1/Deployment.cs]: "in" sounds good to me. I can make the update to the proto(s) as part of the other changes I'm making now, if that helps.

File: apis/deployment_v1alpha1/Deployment.cs [hunk 2]
```diff
@@ -1781,6 +1786,20 @@ public sealed partial class ListDeploymentsRequest : pb::IMessage<ListDeployment
       }
     }
 
+    /// <summary>Field number for the "deployment_stopped_status_filter" field.</summary>
+    public const int DeploymentStoppedStatusFilterFieldNumber = 6;
+    private global::Improbable.SpatialOS.Deployment.V1Alpha1.ListDeploymentsRequest.Types.DeploymentStoppedStatusFilter deploymentStoppedStatusFilter_ = 0;
+    /// <summary>
+    /// Filter for all deployments, deployments that are stopped or deployments that are not stopped.
```
[the remark this answers is not in the document]
Comment (anne-edwards) [reply apis/deployment_v1alpha1/Deployment.cs]: How about:

> Filter for all deployments, or filter by whether deployments are running or not.

(Assuming that's what it means - that you can have it return (a) everything, (b) only running deployments, or (c) only stopped deployments)

Again, happy to add this to my list of changes.

File: apis/deployment_v1alpha1/Deployment.cs [hunk 3]
```diff
@@ -2606,10 +2686,21 @@ public sealed partial class UpdateDeploymentRequest : pb::IMessage<UpdateDeploym
     public const int UpdateMaskFieldNumber = 2;
     private global::Google.Protobuf.WellKnownTypes.FieldMask updateMask_;
     /// <summary>
-    /// A mask which you can provide for partial updates of the resource.
-    /// Only the fields specified in the mask are modified on the response.
+    /// Use `update_mask` to provide partial object updates.
+    ///
+    /// We only support these fields, as in the proto serialised way:
```
[the remark this answers is not in the document]
Comment (lx223) [reply apis/deployment_v1alpha1/Deployment.cs]: It just means how protobuf serialises them to strings. Do you have suggestions that how we can make it clearer?

File: apis/deployment_v1alpha1/Deployment.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (RNabel) [reply apis/deployment_v1alpha1/Deployment.cs]: I have no idea what the actual functionality of this is, so I'm guessing @lx223 is the person for this :)

File: apis/deployment_v1alpha1/Deployment.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (RNabel) [reply apis/deployment_v1alpha1/Deployment.cs]: So it expects a proto-serialized proto message with any of the following fields? In that case, I'd go with "We support the following fields, which must be passed in protobuf serialized form" or similar. But I don't feel I have quite enough context here.

[the remark this answers is not in the document]
Comment (lx223) [reply apis/deployment_v1alpha1/Deployment.cs]: Your phrasing sounds good as well. 

cc @anne-edwards

File: apis/deployment_v1alpha1/Deployment.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (lx223) [reply apis/deployment_v1alpha1/Deployment.cs]: Max from Compute added this feature.

Empty `filter` field filters nothing, i.e. list returns all deployments regardless of their statuses. When it is not empty, it filters out deployments of matching statuses. 

I'll let him know about this as well.

[the remark this answers is not in the document]
Comment (anne-edwards) [reply apis/deployment_v1alpha1/Deployment.cs]: Thanks. Aside from leaving it blank to return all deployments, what can you put in this field? I couldn't find the exact strings. I'm guessing something like `stopped` or `running`, and whichever you specify, it only returns those deployments?

[the remark this answers is not in the document]
Comment (majolo) [reply apis/deployment_v1alpha1/Deployment.cs]: Yeah @anne-edwards "(Assuming that's what it means - that you can have it return (a) everything, (b) only running deployments, or (c) only stopped deployments)" is exactly what it does. The field is a proto type and can be:

`var ListDeploymentsRequest_DeploymentStoppedStatusFilter_name = map[int32]string{
	0: "ALL_DEPLOYMENTS",
	1: "NOT_STOPPED_DEPLOYMENTS",
	2: "STOPPED_DEPLOYMENTS",
}`

I assume the platform SDK generates in such a way that the IDE makes this clear?

[the remark this answers is not in the document]
Comment (majolo) [reply apis/deployment_v1alpha1/Deployment.cs]: Also to be clear the reason its not "running" and "stopped" is that NOT_STOPPED means literally any other state and not just "running". (e.g. "being created")

**Comment by lx223** [comment]
Obsolete

**Pull request closed without merging**